content/sensu-go/5.5/reference/agent.md

```diff
@@ -112,6 +112,55 @@ HTTP/1.1 202 Accepted
 
 _PRO TIP: You can use the agent API `/events` endpoint to create proxy entities by including a `proxy_entity_name` attribute within the `check` scope._
 
+#### Detecting silent failures
+
+You can use the Sensu agent API in combination with the check time-to-live attribute (TTL) to detect silent failures, creating what's commonly referred to as a "dead man's switch" (source: [Wikipedia][20]).
+By using check TTLs, Sensu is able to set an expectation that a Sensu agent will publish additional events for a check within the period of time specified by the TTL attribute.
+If a Sensu agent fails to publish an event before the check TTL expires, the Sensu backend creates an event with a status of `1` (warning) to indicate the expected event was not received.
+For more information on check TTLs, see the [the check reference][44].
+
+A great use case for the Sensu agent API is to enable tasks which run outside of Sensu's check scheduling to emit events. Using the check TTL attribute, these events create a dead man's switch, ensuring that if the task fails for any reason, the lack of an "all clear" event from the task will notify operators of a silent failures which might otherwise be missed.
+If an external source sends a Sensu event with a check TTL to the Sensu agent API, Sensu expects another event from the same external source before the TTL expires.
+
+The following is an example of external event input via the Sensu agent API using a check TTL to create a dead man's switch for MySQL backups.
+Imagine that a MySQL backup script runs periodically, and that we expect the job to take a little less than 7 hours to complete. In the case where the job completes successfully, we'd like a record of it, but don't need to be alerted. If the job fails for some reason, or continues running past the expected 7 hours, we'd like to be alerted.  Using the following code, our script sends an event which tells the Sensu backend to expect an additional event with the same name within 7 hours of the first event.
+
+{{< highlight shell >}}
+curl -X POST \
+-H 'Content-Type: application/json' \
+-d '{
+  "check": {
+    "metadata": {
+      "name": "mysql-backup-job"
+    },
+    "status": 0,
+    "output": "mysql backup initiated",
+    "ttl": 25200
+  }
+}' \
+http://127.0.0.1:3031/events
+{{< /highlight >}}
+
+With this initial event submitted to the Agent API, we have recorded in the backend that our script started, and configured the dead man's switch so that we'll be alerted if the job fails or runs too long. Although it is possible for our script to gracefully handle errors and emit additional monitoring events, this approach allows us to worry less about handling every possible error case, as the lack of additional check result before the 7 hour period elapses will result in an alert.
```

Comment:
```suggestion
With this initial event submitted to the agent API, we have recorded in the Sensu backend that our script started, and we've configured the dead man's switch so that we'll be alerted if the job fails or runs too long. Although it is possible for our script to handle errors gracefully and emit additional monitoring events, this approach allows us to worry less about handling every possible error case, as the lack of additional events before the 7 hour period elapses results in an alert.
```